lib/elixir/lib/exception.ex

```diff
@@ -787,7 +787,14 @@ defmodule File.Error do
   defexception [:reason, :path, action: ""]
 
   def message(exception) do
-    formatted = IO.iodata_to_binary(:file.format_error(exception.reason))
+    formatted =
+      case {exception.action, exception.reason} do
```

Comment:
We are accessing `exception` fields multiple times, what if we decompose exception on first line:
```elixir
%{action: action, reason: reason, path: path} = exception
formatted =
  case {action, reason} do
```
?